### PR TITLE
✨ Feat(#85ztre9m5): added lives logic

### PR DESCRIPTION
**Pull request by Ronishaaa**

### CODE REVIEW

**Comment by rubek-joshi** [comment]
Task linked: [CU-85ztre9m5 Game Logic](https://app.clickup.com/t/85ztre9m5)

File: src/context/GameContext.tsx [hunk 1]
```diff
@@ -1,2 +1,131 @@
...
+import { createContext, useState } from "react";
+import { Question } from "screens/Quiz/Question";
+import { Result } from "screens/Quiz/Result";
+import { GameOver } from "screens";
+import mockData from "../screens/Quiz/cancer-findings-data.json";
+
+interface GameContextProps {
+  actualAnswer: boolean;
+  screens: JSX.Element[];
+  answer: boolean;
```
Comment (rubek-joshi) [inline src/context/GameContext.tsx:12]: src, answer, question, explanation should be nested

Something like this:
```ts
interface Quiz {
  question: string;
  answer: boolean
}

interface GameContextProps {
  highScore: number;
  currentQuestion: Quiz;
}
```

File: src/context/GameContext.tsx [hunk 2]
```diff
@@ -1,2 +1,131 @@
...
+  currentScore: number;
+  highScore: number;
+  activeIndex: number;
+  remainingLives: number;
+  question: string;
+  explanation: string;
+  src: string;
+  handleOnClick: (currentScore: number) => void;
```
Comment (rubek-joshi) [inline src/context/GameContext.tsx:20]: `handleOnClick` function name is too vague

File: src/hooks/useGameContext.tsx [hunk 3]
```diff
@@ -1 +1,13 @@
 // TODO: implement useGameContext hook that also throws error if not used within provider
```
Comment (rubek-joshi) [inline src/hooks/useGameContext.tsx:1]: Remove comment

File: src/main.tsx [hunk 4]
```diff
@@ -1,11 +1,14 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App.tsx'
-import './styles/_main.scss'
+import React from "react";
+import ReactDOM from "react-dom/client";
+import App from "./App.tsx";
+import "./styles/_main.scss";
+import { GameContextProvider } from "context/GameContext.tsx";
```
Comment (rubek-joshi) [inline src/main.tsx:5]: import should be:
```ts
import { GameContextProvider } from "context";
```

File: src/screens/Quiz/Question.tsx [hunk 5]
```diff
@@ -1,17 +1,17 @@
-import deodorant from "assets/questionImg/deodorant.svg";
+// import deodorant from "/questionImg/deodorant.svg";
 import styles from "./Question.module.scss";
+import { useGameContext } from "hooks/useGameContext";
```
Comment (rubek-joshi) [inline src/screens/Quiz/Question.tsx:3]: import should be:
```ts
import { useGameContext } from "hooks";
```

File: src/screens/Quiz/components/Lives.tsx [hunk 6]
```diff
@@ -1,11 +1,13 @@
 import styles from "./Lives.module.scss";
+import { useGameContext } from "hooks/useGameContext";
```
Comment (rubek-joshi) [inline src/screens/Quiz/components/Lives.tsx:2]: import should be:
```ts
import { useGameContext } from "hooks";
```

File: src/screens/Quiz/components/Lives.tsx [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 
 const { life, lifeLost } = styles;
 
-const REMAINING_LIVES = 2;
-const TOTAL_LIVES = 3;
-
 export const Lives = () => {
+  const { remainingLives } = useGameContext();
+
+  // const REMAINING_LIVES = remainingLives;
```
Comment (rubek-joshi) [inline src/screens/Quiz/components/Lives.tsx:9]: remove comment

**Review by rubek-joshi — CHANGES_REQUESTED** [review]

File: src/context/GameContext.tsx [hunk 8]
```diff
@@ -1,2 +1,131 @@
...
+  const [answer, setAnswer] = useState(true);
+
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  const [currentScore, setCurrentScore] = useState(0);
+
+  const initialHighScore = parseInt(localStorage.getItem("highScore") ?? "0");
+
+  //   const highScoreString = localStorage.getItem("highScore");
+  //   const initialHighScore = highScoreString ? parseInt(highScoreString) : 0;
```
Comment (prakash-2021) [inline src/context/GameContext.tsx:55]: remove comments

File: src/screens/GameOver/index.tsx [hunk 9]
```diff
@@ -1,3 +1,13 @@
+import { useGameContext } from "hooks/useGameContext";
```
Comment (prakash-2021) [inline src/screens/GameOver/index.tsx:1]: import should not be like this

File: src/screens/GameOver/index.tsx [hunk 10]
```diff
@@ -1,3 +1,13 @@
...
+
 export const GameOver = () => {
-  return <div>GameOver</div>;
+  const { explanation, currentScore, highScore } = useGameContext();
+  return (
+    <div>
```
Comment (prakash-2021) [inline src/screens/GameOver/index.tsx:6]: useless div 
use <></> instead

File: src/screens/Quiz/Question.tsx [hunk 11]
```diff
@@ -1,17 +1,17 @@
...
 
 export const Question = () => {
+  const { question, src } = useGameContext();
+
   return (
     <div className="flex justify-center">
       <div className={styles.container}>
         <div className={styles.innerContainer}>
-          <img className={styles.image} src={deodorant} alt="deodorant" />
+          <img className={styles.image} src={`/questionImg/${src}`} alt={src} />
```
Comment (prakash-2021) [inline src/screens/Quiz/Question.tsx:12]: alterantive text = src ?

File: src/screens/Quiz/Result.tsx [hunk 12]
```diff
@@ -1,26 +1,33 @@
 import Correct from "assets/resultEmoji/right.png";
 import Wrong from "assets/resultEmoji/wrong.png";
 import styles from "./Result.module.scss";
+import { useGameContext } from "hooks/useGameContext";
```
Comment (prakash-2021) [inline src/screens/Quiz/Result.tsx:4]: import same as above

File: src/screens/Quiz/components/GameActions.tsx [hunk 13]
```diff
@@ -1,14 +1,38 @@
-// import no from "assets/buttonIcon/no.svg";
-// import yes from "assets/buttonIcon/yes.svg";
+import no from "assets/buttonIcon/no.svg";
```
Comment (prakash-2021) [inline src/screens/Quiz/components/GameActions.tsx:1]: import same as above

File: src/screens/Quiz/components/GameActions.tsx [hunk 14]
```diff
@@ -1,14 +1,38 @@
...
+import yes from "assets/buttonIcon/yes.svg";
 import Next from "assets/buttonIcon/next-arrow.svg";
-import { Button } from "components";
+import { Button, IconButton } from "components";
+import { useGameContext } from "hooks/useGameContext";
```
Comment (prakash-2021) [inline src/screens/Quiz/components/GameActions.tsx:5]: import

File: src/context/GameContext.tsx [hunk 15]
```diff
@@ -1,2 +1,131 @@
...
+
+  const [remainingLives, setRemainingLives] = useState(3);
+
+  const screens = [
+    <Question key={1} />,
+    <Result key={2} isCorrect={data[activeQuestionIndex].fact} />,
+    <GameOver key={3} />,
+  ];
+
+  const handleOnClick = (currentScore: number) => {
```
Comment (prakash-2021) [inline src/context/GameContext.tsx:67]: rename to meaningful name

**Review by prakash-2021 — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx [hunk 16]
```diff
@@ -9,7 +9,16 @@ interface Props {
 
 export const Button = ({ label = "", onClick, icon }: Props) => {
   return (
-    <button className={styles.btnNext} onClick={onClick}>
+    <button
+      className={
+        label == "Next"
```
Comment (rubek-joshi) [inline src/components/Button.tsx:14]: Don't use labels and values to set button styles

Comment (rubek-joshi) [inline src/components/Button.tsx:14]: Don't use labels and values to set button styles.

**Review by rubek-joshi — CHANGES_REQUESTED** [review]

**Pull request merged**